### DISCUSSION TITLE
Problems with encoding

### DISCUSSION THREAD

**Discussion by yuzhaninan** [benclaussen/NetboxPS#33] [General]
When executing GET requests, the response is received in UTF 8 encoding, but the Cyrillic character "х" cannot be processed correctly. Thus, it is necessary to use in the POST and PATCH methods the conversion of this character to another. How can this be won?
Have to use
![image](https://user-images.githubusercontent.com/107910784/174734343-b15b7e58-a448-448d-b61f-e059a8b3f98c.png)

**Comment by alagoutte** [comment]
What PowerShell release do you are using ?

**Reply by yuzhaninan** [reply]
5.1

**Reply by alagoutte** [reply]
do you have try with PS 7 ?

**Reply by yuzhaninan** [reply]
Not everything for 5.1 is compatible with 7
Implemented part of the code in Powershell 7
The result is the same. I read that in 7 they fixed the choice of encoding for the Invoke-RestMethod cmdlet response. (in 5.1 hardcoded UTF-8)
In Powershell 7 

'Ð¦Ñ^ÃÑ^ÅÐ½Ð¸ÐºÐ¸Ð´Ð¸ Ð®Ñ^ÀÐ¸Ð¹ Ð^ßÐµÑ^ÂÑ^ÀÐ¾Ð²Ð¸Ñ^Ç' | Fix-Encode
Цухникиди Юрий Петрович
$TO = 'UTF-8' ; $From = 'ISO-8859-1'
$response = 'Ð¦Ñ^ÃÑ^ÅÐ½Ð¸ÐºÐ¸Ð´Ð¸ Ð®Ñ^ÀÐ¸Ð¹ Ð^ßÐµÑ^ÂÑ^ÀÐ¾Ð²Ð¸Ñ^Ç'
[System.Text.Encoding]::GetEncoding($TO).GetString(
[System.Text.Encoding]::GetEncoding($From).GetBytes(
($response | ConvertTo-Json -Depth 10)
))
"Цу�\u0085никиди Юрий Петрович"

**Reply by alagoutte** [reply]
:-(

**Reply by FluffyBeaver** [reply]
I had the same problem with Cyrillic and Greek charsets.
I fixed it by replacing `Invoke-RestMethod` with `Invoke-WebRequest` in Functions/Helpers/InvokeNetboxRequest.ps1.
Like that:
```
$splat = @{
        'Method'      = $Method
        'Uri'         = $URI.Uri.AbsoluteUri # This property auto generates the scheme, hostname, path, and query
        'Headers'     = $Headers
        'TimeoutSec'  = $Timeout
        'ContentType' = "application/json; charset=$([System.Text.Encoding]::Default.WebName)" # <--- This line is modified 
        'ErrorAction' = 'Stop'
        'Verbose'     = $VerbosePreference
    }
...

# $result = Invoke-RestMethod @splat # <--- instead this line I use three lines below
$Response = Invoke-WebRequest @splat
$reader = New-Object System.IO.StreamReader($response.RawContentStream, [System.Text.Encoding]::UTF8)
$result = $reader.ReadToEnd() | ConvertFrom-Json
```
